### PR TITLE
mqtt: Improve handling of multiple PDU parsing

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10262

Issue: 6592

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6592](https://redmine.openinfosecfoundation.org/issues/6592)

Describe changes:
- Parse PDU instead of entire stream

Updates:
- Ensure frame created with proper size

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1609
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by catenacyber** [comment]
Guess SV PR needs an update, see my remarks there

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17948

File: rust/src/mqtt/mqtt.rs [hunk 1]
```diff
@@ -462,14 +462,15 @@ impl MQTTState {
         }
 
         while !current.is_empty() {
-            SCLogDebug!("request: handling {}", current.len());
+            SCLogDebug!("request: handling {}; rem: {}", current.len(), input.len() - current.len());
             match parse_message(current, self.protocol_version, self.max_msg_len) {
                 Ok((rem, msg)) => {
+                    SCLogDebug!("request: handling {}; rem: {}", current.len(), input.len() - current.len());
```
Comment (catenacyber) [inline rust/src/mqtt/mqtt.rs:468]: Why are you duplicating the log message ?
And why are you logging `input.len() - current.len()` as rem (and not `rem.len()`)

**Comment by jlucovsky** [comment]
Continued in #10305

**Pull request closed without merging**